src/main/scala/is/hail/expr/types/PhysicalType.scala

```diff
@@ -1,7 +1,329 @@
 package is.hail.expr.types
 
+import is.hail.annotations._
+import is.hail.asm4s._
+import is.hail.expr.ir.EmitMethodBuilder
+import is.hail.utils._
+
 abstract class PhysicalType {
   def virtualType: Type
+
+  def byteSize: Long
+
+  def alignment: Long
+
+  def unsafeOrdering(rightType: PhysicalType, missingGreatest: Boolean): UnsafeOrdering
```

Comment:
Ordering should be a property on Type: the ordering doesn't change if the physical type does.  I feel like this should be on Type but take a left and right PType.  In particular, then we won't have to implement ordering when we add a new PType.

Reply:
Yeah, this didn't really feel like the right place.

Reply:
will remove